graphql/graphql.go

```diff
@@ -376,32 +398,39 @@ func (t *Transaction) From(ctx context.Context, args BlockNumberArgs) (*Account,
 }
 
 func (t *Transaction) Block(ctx context.Context) (*Block, error) {
-	if _, err := t.resolve(ctx); err != nil {
+	_, block, err := t.resolve(ctx)
+	if err != nil {
 		return nil, err
 	}
-	return t.block, nil
+	return block, nil
 }
 
 func (t *Transaction) Index(ctx context.Context) (*int32, error) {
-	if _, err := t.resolve(ctx); err != nil {
+	_, block, err := t.resolve(ctx)
+	if err != nil {
 		return nil, err
 	}
-	if t.block == nil {
+	// Pending tx
+	if block == nil {
 		return nil, nil
 	}
+	t.mu.Lock()
```

Comment:
If you just want to read an `int32` here, you could use atomic, making the `t.index` into an  `atomic.Int32`. 
```
index := t.index.Load()
```
 But, thinking lcoser on it... How could this possibly race? You aleady did the `resolve`, and no if another thread is now in the `resolve`, it is _not_ going to actually overwrite anything, just see that `t.tx` is non-nil, and return. 

So I think it is safe to remove the locking here. (?)


OTOH, even if it's safe to remove it, might be better to leave it there, since the safety of it is not obvious at a glance, might be better to not have code which looks wrong.